transport/endpoint.go

```diff
@@ -0,0 +1,12 @@
+package transport
+
+import "github.com/aws/smithy-go"
+
+// Endpoint is a Smithy endpoint.
+type Endpoint struct {
+	URI string
```

Comment:
**Question**: Same question, why not just use [net/url#URL](https://pkg.go.dev/net/url#URL)?

Reply:
discussed offline, replaced URI string type with Go URL type. the rationale being its simpler to represent something that more closely resembles reality.
replaced `Fields` with Go net Headers. the rationale being the Go net Headers do the same thing without us having to maintain our own version.